packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -215,7 +215,7 @@ Future<XcodeBuildResult> buildXcodeProject({
   final bool hasFlutterPlugins = injectPlugins();
 
   if (hasFlutterPlugins)
-    await _runPodInstall(appDirectory, flutterFrameworkDir(mode));
+    await cocoaPods.processPods(appDirectory, flutterFrameworkDir(mode));
```

Comment:
We should always run `pod install` if the project has a Podfile, so maybe `processPods()` should take `hasFlutterPlugins` as argument, create the Podfile if required (hasPlugins && !podfile.exists), and finally run `pod install` if the project has a Podfile (either pre-existing, or the one we just created).